base/genericmemory.jl

```diff
@@ -5,7 +5,19 @@
 """
     GenericMemory{kind::Symbol, T, addrspace=Core.CPU} <: AbstractVector{T}
 
-One-dimensional dense array with elements of type `T`.
+One-dimensional, fixed-size, dense array with elements of type `T`.
+
+`kind` can currently be either `:not_atomic` or `:atomic`. For details on what `:atomic` implies, see [`AtomicMemory`](@ref)
+
+`addrspace` can currently only be set to Core.CPU. It is designed to  to permit extension by other systems
+such as GPUs, which might define values such as:
+```
+module CUDA
+const Generic = bitcast(Core.AddrSpace{CUDA}, 0)
+const Global = bitcast(Core.AddrSpace{CUDA}, 1)
```

Comment:
```suggestion
const Generic = Core.AddrSpace{CUDA}, 0x00)
const Global = Core.AddrSpace{CUDA}, 0x01)
```
This code doesn't work since a) `bitcast` is not public, and b) the integer must be a 1-byte integer for bitcast to work

Reply:
I think this was unresolved with the force push.

Reply:
oops. Will re-resolve once https://github.com/JuliaLang/julia/pull/54681 merges.